lib/chef/resource.rb

```diff
@@ -296,17 +296,36 @@ def method_missing(method_symbol, *args, &block)
       end
     end
 
+    def identical_resource?(prior_resource)
+      skipped_ivars = [ :@source_line, :@cookbook_name, :@recipe_name, :@params, :@elapsed_time ]
```

Comment:
IIRC `@params` is used by definitions. I once tried to rip it out but there was a bunch of really ugly stuff tied up in it.